signature/docker.go

```diff
@@ -49,16 +50,25 @@ func SignDockerManifest(m []byte, dockerReference string, mech SigningMechanism,
 // VerifyDockerManifestSignature checks that unverifiedSignature uses expectedKeyIdentity to sign unverifiedManifest as expectedDockerReference,
 // using mech.
 func VerifyDockerManifestSignature(unverifiedSignature, unverifiedManifest []byte,
-	expectedDockerReference string, mech SigningMechanism, expectedKeyIdentity string) (*Signature, error) {
+	expectedDockerReference string, mech SigningMechanism, expectedKeyIdentity string) (sig *Signature, err error) {
```

Comment:
Don’t use named return values please; they might hide errors (e.g. failing to set `err` on an error path), and interact in surprising ways with `defer` ( https://github.com/containers/storage/pull/1483#discussion_r1154252787  ).

(Yes, paranoia is rampant throughout the `signature` subpackage…)